### PR TITLE
Make the configuration file be driven by command line arguments

### PR DESCRIPTION
**Pull request by gugibugy**
- This is in contrast of the current approach which relies on updating hardcoded string paths inside of mpl_config.py. 
- Most of the changes comprise updating function calls and constructors to take in a config object which is then used to reference the relevant attributes of the class instead of accessing stacking variables from the config.
- Please pay extra attention to the changes in mpl_infer_tiles_GPU_new.py
    - The predictor constructor was updated to support creating predictors that run on either a CPU or a GPU
    - behavior of how the predictor run method pulls jobs from the input queue was updated to remove infinite "while true" loop and base it on the input_queue being empty. This means that the order of operations in inference_image had to be changed so that the jobs are added to the queue BEFORE the processes are started. 
- There are a lot of misc changes due to running the Black formatter on the files that updated

### CODE REVIEW

File: maple_workflow.py [hunk 1]
```diff
@@ -45,14 +45,13 @@ def tile_image(input_img_name):
     ----------
     input_img_name : Name of the input image
     """
```
Comment (kaylahardie) [inline maple_workflow.py:48]: super nit but you could add an arg docstring for config here and some of the other function docs in this file

File: mpl_infer_tiles_GPU_new.py [hunk 2]
```diff
@@ -21,17 +23,28 @@
...
 
+class Predictor(multiprocessing.Process):
+    def __init__(
+        self,
+        config: MPL_Config,
+        input_queue: multiprocessing.JoinableQueue,
+        use_gpu: bool,
+        process_counter: int,
+        POLYGON_DIR,
+        weights_path: str,
```
Comment (kaylahardie) [inline mpl_infer_tiles_GPU_new.py]: can you explain how this weights_path is different from the weights path defined in the config?

File: mpl_infer_tiles_GPU_new.py [hunk 3]
```diff
@@ -21,17 +23,28 @@
...
-                          output_shp_root,
-                          x_resolution,
-                          y_resolution,len_imgs,image_name):
 
+class Predictor(multiprocessing.Process):
+    def __init__(
+        self,
+        config: MPL_Config,
+        input_queue: multiprocessing.JoinableQueue,
+        use_gpu: bool,
```
Comment (kaylahardie) [inline mpl_infer_tiles_GPU_new.py]: similarly - could we look at the number of gpus variable in the config?

File: mpl_infer_tiles_GPU_new.py [hunk 4]
```diff
@@ -69,146 +78,179 @@ def run(self):
         # Useful if you're training a model on the same
         # machine, in which case use CPU and leave the
         # GPU for training.
-        DEVICE = "/gpu:%s"%(self.gpu_id)  # /cpu:0 or /gpu:0
-        os.environ['CUDA_VISIBLE_DEVICES'] = "{}".format(self.gpu_id)
+        if self.use_gpu:
+            os.environ["CUDA_VISIBLE_DEVICES"] = "{}".format(self.process_counter)
```
Comment (kaylahardie) [inline mpl_infer_tiles_GPU_new.py:80]: nit could add a comment about what a process counter is?

File: mpl_infer_tiles_GPU_new.py [hunk 5]
```diff
@@ -69,146 +78,179 @@ def run(self):
...
+def inference_image(
+    config: MPL_Config,
+    POLYGON_DIR: str,
+    weights_path: str,
+    output_shp_root: str,
+    file1: str,
+    file2: str,
+    image_name: str,
+):
+    f1 = h5py.File(file1, "r")
```
Comment (kaylahardie) [inline mpl_infer_tiles_GPU_new.py:180]: what are these two files exactly?

**Review by kaylahardie — APPROVED** [review]
Thanks so much David this is great! In the future, it'd be easier to review if you linted the files in a separate PR (which would be a no-op) so that the changes in this PR are easier to review/more scoped

File: mpl_infer_tiles_GPU_new.py (hunk 2 above)
Comment (gugibugy) [reply mpl_infer_tiles_GPU_new.py]: I can't cause they are the same!!! Great catch, I'll remove the redundancy!

File: mpl_infer_tiles_GPU_new.py (hunk 5 above)
Comment (gugibugy) [reply mpl_infer_tiles_GPU_new.py:180]: My understanding is that these two files are what Amal was referring to in the deep dive meeting when he said they split up the images into two files: one for holding the image data and the other one holds information about how to stitch the images back together to go from tiles to full satellite scene afterwards.

File: maple_workflow.py (hunk 1 above)
Comment (gugibugy) [reply maple_workflow.py:48]: I tried a few descriptions and this is what i ended up with. Lmk if you have a better suggestion!

File: mpl_infer_tiles_GPU_new.py (hunk 4 above)
Comment (gugibugy) [reply mpl_infer_tiles_GPU_new.py:80]: added the comment in the constructor where we set the property.

File: mpl_infer_tiles_GPU_new.py (hunk 3 above)
Comment (gugibugy) [reply mpl_infer_tiles_GPU_new.py]: Done. Lmk what you think!

File: maple_workflow.py (hunk 1 above)
Comment (kaylahardie) [reply maple_workflow.py:48]: what you have lgtm

File: mpl_infer_tiles_GPU_new.py [hunk 6]
```diff
@@ -175,7 +175,6 @@ def run(self):
 def inference_image(
     config: MPL_Config,
     POLYGON_DIR: str,
```
Comment (kaylahardie) [inline mpl_infer_tiles_GPU_new.py]: Another thought - should the POLYGON_DIR also be a part of the config?

File: mpl_infer_tiles_GPU_new.py (hunk 3 above)
Comment (kaylahardie) [reply mpl_infer_tiles_GPU_new.py]: Looks great!

File: mpl_infer_tiles_GPU_new.py (hunk 2 above)
Comment (kaylahardie) [reply mpl_infer_tiles_GPU_new.py]: Thanks for removing them :)

**Comment by kaylahardie** [comment]
Added one more question/open comment

**Review by kaylahardie — APPROVED** [review]

File: mpl_infer_tiles_GPU_new.py (hunk 6 above)
Comment (gugibugy) [reply mpl_infer_tiles_GPU_new.py]: Turns out it wasn't used. Removed from code paths.

**Pull request merged**